### PR TITLE
samba.sh: only restart samba service if service exists

### PR DESCRIPTION
**Pull request by gizmo98**
Fixes misleading restart fail message https://github.com/RetroPie/RetroPie-Setup/issues/3616

### CODE REVIEW

File: scriptmodules/supplementary/samba.sh [hunk 1]
```diff
@@ -41,7 +41,8 @@ _EOF_
 }
 
 function restart_samba() {
-    service samba restart || service smbd restart
+    [[ ! -z $(systemctl list-units --full -all | grep smbd.service) ]] && service smbd restart
+    [[ ! -z $(systemctl list-units --full -all | grep samba.service) ]] && service samba restart
```
Comment (joolswills) [inline scriptmodules/supplementary/samba.sh:45]: Not sure I like the duplication. Do we really need to check - can't we just hide the error ?

``
    local service
    for service in smbd samba; do
        systemctl restart $service 2>/dev/null
    done
``

If the check is needed that could be done in the loop I guess without duplicating the command.

Comment (gizmo98) [reply scriptmodules/supplementary/samba.sh:45]: Redirect output to /dev/null should work as well.

**Comment by cmitu** [comment]
My idea when keeping the linked issue was to remove `samba` as service name and do something like:
```
systemctl -q is-active smbd.service && systemctl -q reload-or-restart smbd.service
```

**Comment by gizmo98** [comment]
@cmitu this does only work if there is no samba.service. Debian stretch has smbd.service and samba.service. Do we remove backwards compatiblity for stretch? Debian bullseye and bookworm have only smbd.service. So we could add a debian version check for samba.service.
https://packages.debian.org/search?mode=path&suite=bookworm&section=all&arch=any&searchon=contents&keywords=samba.service

**Comment by cmitu** [comment]
> Debian stretch has smbd.service and samba.service. Do we remove backwards compatiblity for stretch?

On _stretch_, the `samba.service` unit file is a `/dev/null` symlink - see https://sources.debian.org/src/samba/2%3A4.5.16%2Bdfsg-1%2Bdeb9u2/debian/samba.links/ and [here](https://sources.debian.org/src/samba/2%3A4.5.16%2Bdfsg-1%2Bdeb9u2/debian/samba.install/) for the files included in the package).

I don't think we'll loose compatiblity with _stretch_ by removing support for it, restarting `smbd` works on _stretch_ and should produce the same result.

**Comment by gizmo98** [comment]
Ok. Then your solution looks much better. I will close this PR.

**Pull request closed without merging**